contrib/pylightning/lightning/plugin.py

```diff
@@ -99,7 +99,7 @@ class Plugin(object):
 
     """
 
-    def __init__(self, stdout=None, stdin=None, autopatch=True):
+    def __init__(self, stdout=None, stdin=None, autopatch=True, dynamic=True):
```

Comment:
This was commented about earlier, but is it not more conservative to default to `dynamic=False` ?

Or is it now expected that old plugins need to update anyway to this new behavior?
Most of the plugins don't even have tests!

Reply:
> This was commented about earlier, but is it not more conservative to default to `dynamic=False` ?
> 
> Or is it now expected that old plugins need to update anyway to this new behavior?
> Most of the plugins don't even have tests!

This was discussed above. This does not change the plugin behavior at all, it just makes it possible for lightningd to restart it : this is completely transparent plugin side (hence the new init field).

Reply:
That said, a plugin that is not intended to be stopped at runtime can set dynamic to `False` to ensure that an user won't *inadvertently* stop it via RPC.

Reply:
> This does not change the plugin behavior at all

Sorry to be blunt, but if the behavior is not changed, then why is there an extra parameter?
What is wrong with defaulting to `dynamic=False` ?

Maybe to you its trivial, but how can we know for sure that current (known or unknown) plugins work properly when re-starting with a running c-lightning?

Reply:
> Sorry to be blunt, but if the behavior is not changed, then why is there an extra parameter?
What is wrong with defaulting to dynamic=False ?

This is a new field passed in the `getmanifest` call, [used to give informations to lightningd about plugins](https://lightning.readthedocs.io/PLUGINS.html#the-getmanifest-method). This allows a plugin creator to pass whatever he wants.

> What is wrong with defaulting to dynamic=False ?

Pylightning users not setting it.

> Maybe to you its trivial, but how can we know for sure that current (known or unknown) plugins work properly when re-starting with a running c-lightning?

 Restarting a plugin means stopping and starting it : on the `stop` command, the plugin will be killed. On the `start` command an interface similar (but asynchronous) to which is setup on startup is setup by `lightningd` : a plugin must behave properly at `start` or it means it wouldn't even at startup.
The only case this could lead to an issue is a plugin which has to be loaded (for whatever reason, i.e. backup the first db calls) at startup *only* : that was `myplugin.dynamic = False` is for :-)

Reply:
Ok thanks for the clarification, now I also read @cdecker's comment:

> Should be ok for now, I'll fixup the various plugins in lightningd/plugins once this gets merged.